### PR TITLE
Add application-title-bar widget-specific options

### PR DESCRIPTION
**Pull request by HeitorAugustoLN**
Implements third-party widget-specific options for application-title-bar as suggested in #229

### CODE REVIEW

File: modules/widgets/application-title-bar.nix [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+          default = null;
+          description = "The horizontal alignment of the widget.";
+          apply = convertHorizontalAlignment;
+        };
+        verticalAlignment = mkOption {
+          type = types.nullOr (types.enum [ "top" "center" "bottom" "baseline" ]);
+          default = null;
+          description = "The vertical alignment of the widget.";
+          apply = convertVerticalAlignment;
+        };
```
Comment (magnouvean) [inline modules/widgets/application-title-bar.nix]: These allow for `null` in the type (in fact it defaults to `null`), but it crashes when these options are `null`. Either allow for `null` values in `convertVericalAlignment`/`convertHorizontalAlignment` or don't allow them to take the value `null` at all (and set some default values)

**Comment by magnouvean** [comment]
Other than that it looks OK for the most part I reckon. I don't really have time to test all these options though, but maybe I'll test a few of them tomorrow and see if they generally work. In the future we should maybe also add an option to add the widget automatically to `extraWidgets`, but the interface isn't really there yet so that can be done in another pr.

File: modules/widgets/application-title-bar.nix (hunk 1 above)
Comment (HeitorAugustoLN) [reply modules/widgets/application-title-bar.nix]: Fixed. I didn't notice it when I was writing it.

**Comment by HeitorAugustoLN** [comment]
> Other than that it looks OK for the most part I reckon. I don't really have time to test all these options though, but maybe I'll test a few of them tomorrow and see if they generally work.

As far as I tested, everything should be working right now.

> In the future we should maybe also add an option to add the widget automatically to `extraWidgets`, but the interface isn't really there yet so that can be done in another pr.

That is a good idea, I could try doing it later.

**Comment by magnouvean** [comment]
Nice that has done the trick. I'll merge right away then as this looks good. Great work!

**Comment by HeitorAugustoLN** [comment]
Thanks!

**Pull request merged**